stm32: ethernet related changes

erwango opened the pull request:
Few ethernet changes for stm32:

*Add default configuration for ethernet support
*Provide minor updates to stm32 ethernet driver

erwango committed:
board: nucleo_f429zi provide eth driver config

If NETWORKING is activated, enable L2 ethernet and stm32 ethernet driver

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

galak commented on drivers/ethernet/eth_stm32_hal.c:
```diff
@@ -340,8 +341,10 @@ static void eth0_iface_init(struct net_if *iface)
+
+	if (hal_ret != HAL_OK) {
+		SYS_LOG_ERR("HAL_ETH_Init failed: %d\n", hal_ret);
 	}
```
Does it make sense to continue if HAL_ETH_Init isn't HAL_OK?

erwango replied:
Yes, sure, we can exit at this point, makes sense

galak commented on drivers/ethernet/eth_stm32_hal.c:
```diff
@@ -340,8 +341,11 @@ static void eth0_iface_init(struct net_if *iface)
+	if (hal_ret != HAL_OK) {
+		SYS_LOG_ERR("HAL_ETH_Init failed: %d\n", hal_ret);
+		return;
 	}
```
Does it make sense to move this before we call k_thread_create?

erwango replied:
Yeah so we avoid having thread being created and left unused, this is right.

lpereira commented on drivers/ethernet/eth_stm32_hal.c:
```diff
@@ -394,6 +403,8 @@ static struct eth_stm32_hal_dev_data eth0_data = {
 			.ChecksumMode = ETH_CHECKSUM_BY_SOFTWARE,
 			.MediaInterface = ETH_MEDIA_INTERFACE_RMII,
+			.DuplexMode = ETH_MODE_FULLDUPLEX,
+			.Speed = ETH_SPEED_100M,
```
Will this disable auto negotiation of link speed and duplex mode?  Although we're in 2017, I can imagine this being plugged to a crusty switch that will only negotiate a half-duplex 10M link.

erwango replied:
You're right, I missed these are defined after auto negotiation. I'll remove this. Thanks!

erwango committed:
drivers: ethernet: stm32: various small changes

Amend stm32 ethernet driver with small changes:
*Provide HAL_ETH_Init return value in error message,
return on error and move it before thread creation
*Provide computed MAC address in debug message

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
